src/btree/bt_random.c

```diff
@@ -377,7 +377,7 @@ __wt_random_descent(WT_SESSION_IMPL *session, WT_REF **refp, uint32_t flags, WT_
     if (eviction) {
         if (S2C(session)->evict_random_retries) {
             WT_INTL_INDEX_GET(session, (&btree->root)->page, pindex);
-            retry = (int)pindex->entries;
+            retry = WT_MIN((int)pindex->entries, 10000);
```

Comment:
Add a comment here to explain why 10k is chosen?

Reply:
I agree - the number seems very high. I'd think that 100 was enough - if 100 random positions don't find a valid ref, then it's very likely more efficient to just go left-to-right and skip a few pages in-memory, rather than trying to find a rare in-memory page.

Reply:
For in-memory pages, may be after 100 reties, we should scan the whole page index and get all the in-memory children and randomly selects from them.

Reply:
Makes sense to me @quchenhao. I will start working on this, thanks.

Reply:
Implemented [15bab0f](https://github.com/wiredtiger/wiredtiger/pull/10437/commits/15bab0fae33f235bed478030ce91a509ca36657d), yet to add stats.